Kernel/Net/NetworkAdapter.cpp

```diff
@@ -169,4 +169,34 @@ void NetworkAdapter::set_ipv6_netmask(IPv6Address const& netmask)
     m_ipv6_netmask = netmask;
 }
 
+void NetworkAdapter::autoconfigure_link_local_ipv6()
+{
+    auto mac = mac_address();
+    if (mac.is_zero() || !link_up())
+        return;
+
+    // TODO: other IPv6 autoconf modes
+    // TODO: duplicate address detection, this is only a very naive implementation of autoconf
+    auto ipv6_ll = IPv6Address({ 0xfe,
+        0x80,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        mac[0],
+        mac[1],
+        mac[2],
+        mac[3],
+        mac[4],
+        mac[5] });
+    auto netmask = IPv6Address({ 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0, 0, 0, 0, 0, 0, 0, 0 });
+    set_ipv6_address(ipv6_ll);
+    set_ipv6_netmask(netmask);
+    dbgln("autoconfigure_link_local_ipv6: configured address {}", ipv6_ll.to_string());
```

Comment:
not sure if this is different to the other parts of the kernel, but I found it quite common around `NetworkTask.cpp` and `NetworkAdapter.cpp`. In the earlier, each `handle_*` function has its `dbgln_if` prepended with a function name in this fashion

Reply:
Yeah, `dbgln_if`s sometimes do have function names in them.
But none of the `dbgln`s in the boot log have it, so I though it might look a bit out of place there.

Reply:
alright, then imho valid point. this also should be kept as a normal dbgln, I see this as something useful to see for a general user

Reply:
I think it's relevant to contextualize this debug line, since later on there will be multiple places configuring IPv6 addresses without user input, and it's important to easily trace which place that was. Here it's the LLA of course, so I think I will update the function to reflect that.